Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -1699,20 +1699,22 @@ FloatRect RenderLayer::referenceBoxRectForClipPath(CSSBoxType boxType, const Lay
 
 void RenderLayer::updateTransformFromStyle(TransformationMatrix& transform, const RenderStyle& style, OptionSet<RenderStyle::TransformOperationOption> options) const
 {
-    auto referenceBoxRect = snapRectToDevicePixelsIfNeeded(renderer().transformReferenceBoxRect(style), renderer());
-    renderer().applyTransform(transform, style, referenceBoxRect, options);
-
-    // https://drafts.csswg.org/css-anchor-position-1/#anchor-pos
-    // "The positioned element is additionally visually shifted by its snapshotted scroll offset, as if by an additional translate() transform."
+    // https://drafts.csswg.org/css-anchor-position-1/#compensate-for-scroll
+    // "After layout has been performed for abspos, it is additionally
+    // shifted by the default scroll shift, as if affected by a transform
+    // (before any other transforms)."
     if (m_snapshottedScrollOffsetForAnchorPositioning)
         transform.translate(m_snapshottedScrollOffsetForAnchorPositioning->width(), m_snapshottedScrollOffsetForAnchorPositioning->height());
 
+    auto referenceBoxRect = snapRectToDevicePixelsIfNeeded(renderer().transformReferenceBoxRect(style), renderer());
```

Comment:
Your commit message needs to explain why you flipped the ordering of applying the style transform, and then the anchorpos transform. Are there tests that exercise this (e.g. the anchored element having a rotation transform)?

Reply:
I'll factor this out to another patch.